spec.html

```diff
@@ -9480,18 +9473,27 @@ <h1>IntegerIndexedObjectCreate ( _prototype_, _internalSlotsList_ )</h1>
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-integerindexedelementget" aoid="IntegerIndexedElementGet">
+        <h1>IsIntegerIndexValid ( _O_, _index_ )</h1>
+        <p>The abstract operation IsIntegerIndexValid with arguments _O_ and _index_ performs the following steps:</p>
+        <emu-alg>
+          1. Assert: Type(_index_) is Number.
+          1. Assert: _O_ is an Object that has an [[ArrayLength]] internal slot.
+          1. If IsInteger(_index_) is *false*, return *false*.
+          1. If _index_ = *-0*, return *false*.
+          1. If _index_ &lt; 0 or _index_ &ge; _O_.[[ArrayLength]], return *false*.
+          1. Return *true*.
+        </emu-alg>
+      </emu-clause>
+
       <emu-clause id="sec-integerindexedelementget" aoid="IntegerIndexedElementGet">
         <h1>IntegerIndexedElementGet ( _O_, _index_ )</h1>
         <p>The abstract operation IntegerIndexedElementGet with arguments _O_ and _index_ performs the following steps:</p>
         <emu-alg>
-          1. Assert: Type(_index_) is Number.
```

Comment:
why is this assert no longer valid? (ideally all argument types are asserted at the top of every abstract operation)